digdag-core/src/main/java/io/digdag/core/Limits.java

```diff
@@ -8,15 +12,21 @@
     private static final long MAX_ATTEMPTS = Long.valueOf(
             System.getProperty("io.digdag.limits.maxAttempts", "100"));
 
-    // TODO (dano): this should be configurable by config file etc and not just system property
+    private final Config systemConfig;
+
+    @Inject
+    public Limits(Config systemConfig)
+    {
+        this.systemConfig = systemConfig;
+    }
 
-    public static long maxWorkflowTasks()
+    public long maxWorkflowTasks()
     {
-        return MAX_WORKFLOW_TASKS;
+        return systemConfig.get("executor.task_max_run", long.class, MAX_WORKFLOW_TASKS);
```

Comment:
How about memoization of this value in terms of performance?

Reply:
Good point. I am not sure how much it contribute to performance but valuable.

Reply:
Done.